jetty-server/src/test/java/org/eclipse/jetty/server/GracefulStopTest.java

```diff
@@ -657,6 +659,41 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         assertThat(response,Matchers.not(Matchers.containsString("Connection: close")));
         assertTrue(latch.await(10,TimeUnit.SECONDS));
     }
+
+    @Test
+    public void testFailedStart()
```

Comment:
This tests what happens with ContextHandlers. But WebAppContexts are more complex. By default, a WebAppContext will not throw an exception on startup, you have to setThrowUnavailableOnStartupException(true).  Moreover, this test only tests the "manual deployment" scenario - what happens with the automatic WebAppDeployer? Does it swallow exceptions from failed WebAppContexts or pass them upwards?

Reply:
This is only testing what happens if a component of a ContainerLifeCycle fails, which is enough as all the details of a ContextHandler vs WebAppContext is hidden from the container holding it.  Either it does throw or it does not throw (controlled by the setThrowUnavailableOnStartupException) and that is all that matters for this test.      Note also that I have manually tested a WebAppContext, so I know it works for the more complex case.